File: DGV7.0-NON-SAAS/PayHub-Guest-iOS/PayHubGuest/Models/SiteInfoModels.swift
import Foundation

struct SiteInfoResponse: Codable {
    let status: String?
    let data: SiteInfoData?
}

struct SiteInfoData: Codable {
    let siteTitle: String?
    let logoUrl: String?
    let primaryColor: String?
    let secondaryColor: String?
    /// A service key absent from this map must be treated as enabled — it means the admin
    /// never touched Service Control Centre for it, not that it's disabled. Only an explicit
    /// 0 hides a service.
    let services: [String: Int]?
    let currencySymbol: String?
    let support: GuestSupportInfo?

    enum CodingKeys: String, CodingKey {
        case siteTitle = "site_title"
        case logoUrl = "logo_url"
        case primaryColor = "primary_color"
        case secondaryColor = "secondary_color"
        case services
        case currencySymbol = "currency_symbol"
        case support
    }
}

struct GuestSupportInfo: Codable {
    let email: String?
    let phone: String?
    let address: String?
}

File: DGV7.0-NON-SAAS/PayHub-Guest-Android/app/src/main/java/com/payhub/guest/data/model/SiteInfoModels.kt
package com.payhub.guest.data.model

import com.google.gson.annotations.SerializedName

data class SiteInfoResponse(
    val status: String? = null,
    val data: SiteInfoData? = null,
)

data class SiteInfoData(
    @SerializedName("site_title") val siteTitle: String? = null,
    @SerializedName("logo_url") val logoUrl: String? = null,
    @SerializedName("primary_color") val primaryColor: String? = null,
    @SerializedName("secondary_color") val secondaryColor: String? = null,
    // A service key absent from this map must be treated as enabled — it means the admin
    // never touched Service Control Centre for it, not that it's disabled. Only an explicit
    // 0 hides a service.
    val services: Map<String, Int>? = null,
    @SerializedName("currency_symbol") val currencySymbol: String? = null,
    val support: GuestSupportInfo? = null,
)

data class GuestSupportInfo(
    val email: String? = null,
    val phone: String? = null,
    val address: String? = null,
)
